nifi-nar-bundles/nifi-standard-services/nifi-hbase_1_1_2-client-service-bundle/nifi-hbase_1_1_2-client-service/src/main/java/org/apache/nifi/hbase/HBase_1_1_2_LookupService.java

```diff
@@ -0,0 +1,168 @@
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.NavigableMap;
+import java.util.Optional;
+import java.util.Set;
+
+public class HBase_1_1_2_LookupService extends HBase_1_1_2_ClientService implements LookupService {
+    private static final Set<String> REQUIRED_KEYS;
+
+    public static final PropertyDescriptor TABLE_NAME = new PropertyDescriptor.Builder()
+            .name("hb-lu-table-name")
+            .displayName("Table Name")
+            .description("The name of the table where look ups will be run.")
+            .required(true)
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .build();
+    public static final PropertyDescriptor RETURN_CFS = new PropertyDescriptor.Builder()
+            .name("hb-lu-return-cfs")
+            .displayName("Column Families")
+            .description("The column families that will be returned.")
+            .required(true)
+            .addValidator(StandardValidators.NON_BLANK_VALIDATOR)
+            .build();
+    public static final PropertyDescriptor RETURN_QFS = new PropertyDescriptor.Builder()
+            .name("hb-lu-return-qfs")
+            .displayName("Column Qualifiers")
+            .description("The column qualifies that will be returned.")
+            .required(false)
+            .addValidator(Validator.VALID)
+            .build();
+
+    static {
+        REQUIRED_KEYS = new HashSet<String>();
+        REQUIRED_KEYS.add("rowKey");
+    }
+
+    private String tableName;
+    private List<byte[]> families;
+    private List<byte[]> qualifiers;
+
+    private List<PropertyDescriptor> lookupProperties;
+
+    @Override
+    protected void init(ControllerServiceInitializationContext config) throws InitializationException {
+        super.init(config);
+        this.lookupProperties = new ArrayList<>();
+        this.lookupProperties.addAll(properties);
+        this.lookupProperties.add(TABLE_NAME);
+        this.lookupProperties.add(RETURN_CFS);
+        this.lookupProperties.add(RETURN_QFS);
+    }
+
+    @Override
+    public Optional lookup(Map coordinates) throws LookupFailureException {
+        byte[] rowKey = ((String)coordinates.get("rowKey")).getBytes();
+        try {
+            Map<String, Object> values = new HashMap<>();
+            try (Table table = this.connection.getTable(TableName.valueOf(tableName))) {
+                Get get = new Get(rowKey);
+                Result result = table.get(get);
+
+                for (byte[] fam : families) {
+                    NavigableMap<byte[], byte[]>  map = result.getFamilyMap(fam);
+                    for (Map.Entry<byte[], byte[]> entry : map.entrySet()) {
+                        if (qualifiers.contains(entry.getKey()) || qualifiers.size() == 0) {
+                            values.put(new String(entry.getKey()), new String(entry.getValue()));
```

Comment:
It's best to avoid using the String(byte[]) constructor and either explicitly pass StandardCharsets.UTF_8 and document that it expects UTF-8 formatted data or better yet expose a Character Set as a configurable property to use.